addons/mail/static/src/core/web/message_actions_patch.js

```diff
@@ -1,22 +1,106 @@
+import { getNonEditableMentions, parseEmail } from "@mail/utils/common/format";
 import { messageActionsRegistry } from "@mail/core/common/message_actions";
 import { _t } from "@web/core/l10n/translation";
+import { renderToMarkup } from "@web/core/utils/render";
+import { rpc } from "@web/core/network/rpc";
+
+function prepareActionBody(message) {
+    const email_from = message.author?.email ? message.author.email : message.email_from;
+    const [name, email] = parseEmail(email_from);
+    const datetimeFormatted = _t("%(date)s at %(time)s", {
+        date: message.datetime.toFormat("ccc, MMM d, yyyy"),
+        time: message.datetime.toFormat("hh:mm a"),
+    });
+    return renderToMarkup("mail.Message.bodyInForward", {
+        body: getNonEditableMentions(message.body),
+        date: datetimeFormatted,
+        email,
+        message,
+        name: name || email,
+    });
+}
```

Comment:
Not worth sharing function neither: the name is poor, and looks like you made a mistake: `bodyInForward` is shared when in original code it's 2 different templates (`bodyInForward` and `bodyInReply`).

Better keep same shape of code and see it's a move, especially as a `[FIX]` to make the change safer.